observability/probe/aggregate.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package probe
+
+import (
+	"github.com/juju/errors"
+)
+
+type Aggregate struct {
+	// Probes is a map of probes to run as part of this aggregate with the key
+	// corresponding to well known name for the probe.
+	Probes map[string]Prober
+}
+
+func NewAggregate() *Aggregate {
+	return &Aggregate{
+		Probes: make(map[string]Prober),
+	}
+}
+
+// Probe implementes Prober Probe
+func (a *Aggregate) Probe() (bool, error) {
+	return a.ProbeWithResultCallback(func(_ string, _ bool, _ error) {})
```

Comment:
if would be helper to define a type for this function or at least name the parameters here to aid readability, or better, define a NoopCallback and pass that in

Reply:
fair enough.